src/ucp/api/ucp.h

```diff
@@ -2100,6 +2127,32 @@ ucs_status_t ucp_context_query(ucp_context_h context_p,
                                ucp_context_attr_t *attr);
 
 
+/**
+ * @ingroup UCP_MEM
+ * @brief Compare two remote keys
+ *
+ * This routine compares two remote keys.
+ *
+ * It sets the @a result argument to < 0 if rkey1 is lower than rkey2, 0 if they
+ * are equal or > 0 if rkey1 is greater than rkey2. It can be used for sorting.
+ *
+ * @param [in]  context     Handle to @ref ucp_context_h
+ * @param [in]  rkey1       First rkey to compare
+ * @param [in]  rkey2       Second rkey to compare
+ * @param [in]  params      Additional parameters to the comparison
+ * @param [out] result      Result of the comparison
+ *
+ * @return UCS_OK                - @a result contains the comparison result
+ * @return UCS_ERR_UNSUPPORTED   - The operation parameters passed with
```

Comment:
I really don't like the idea of non-supported. If it is non supported is really means keys are not equal. In this case we just should return consistent  `<` or `>' based on some other parameters. E.g u can use pointer values of the underlaying layer, etc.

Reply:
I agree. If the keys are valid we should always be able to compare keys as if they are binary strings.
@tvegas1 in which case we cannot compare?

Reply:
The proposal is to have using rkey compare with proto v1 return unsupported. As this ucp api seems there could be other future cases.

Reply:
Alternative proposal:
- Have all UCT rkey compare implement proper compare and return UCS_OK (the current case). But keep passing them a params and return a ucs_status_t for future proof, add assert preventing not supported error code in UCP after UCT rkey compare.
- In case UCT rkey compare would return an error (future): have UCP rkey compare return it immediately.

For the protov1 check:
- return invalid params
For bad params->field_mask:
- return invalid params

Reply:
why can't we compare unpacked keys with protov1?

Reply:
I would need to check in details, but rkey->cache content might not all be constant, for instance there seems to be rkey resolve at send time. I was under assumption that we did not need v1 support.

Reply:
per offline discussion, will remove ERR_UNSUPPORTED return value from doc

Reply:
fixed, will add comparison for v1 proto too in implementation.

Reply:
@yosefe @tvegas1 - to confirm both proto v1 and proto v2 will be supported ?

Reply:
yes, both will be

Reply:
OK